src/lock_file/resolve/build_dispatch.rs

```diff
@@ -502,4 +512,15 @@ impl BuildContext for LazyBuildDispatch<'_> {
     fn workspace_cache(&self) -> &WorkspaceCache {
         &self.workspace_cache
     }
+
+    fn build_arena(&self) -> &BuildArena<Self::SourceDistBuilder> {
+        // Ensure the build dispatch is initialized
+        self.ensure_build_dispatch_initialized();
```

Comment:
should we also check here that we did not intialized it again?

Reply:
It does that I think :)

Reply:
we have this in interpreter method
```
        if self.lazy_deps.interpreter.get().is_none() {
            self.ensure_build_dispatch_initialized();
        }
```

so I thought maybe we can do it here the same?